### PR TITLE
Deepensemble in cgm-rg

### PR DESCRIPTION
**Pull request by prajurock**
US :- https://dev.azure.com/cgmorg/ChildGrowthMonitor/_workitems/edit/1203
- prediction and uncertainty in tagging tool
scan level: 
**q1-ensemble-warmup-mean** - prediction - uncertainty

- artifact-level
**q1-ensemble-warmup**   prediction -uncertainty

### CODE REVIEW

**Review by shanky0009 — APPROVED** [review]

File: src/download_model.py [hunk 1]
```diff
@@ -71,6 +71,24 @@ def main():
                    input_location=os.path.join('outputs', 'best_model.ckpt'),
                    output_location=REPO_DIR / 'models/depthmapmultiartifactlatefusion')
 
+    RUN_IDS = ['q1-ensemble-warmup_1610544610_eb44bfe2', 'q1-ensemble-warmup_1610547587_7ca932c3',
```
Comment (prajurock) [inline src/download_model.py:74]: Where i should write RUN_IDs at the top?

**Comment by markus-hinsche** [comment]
* Please link the CR in the PR description
* Please ensure the ACs are fulfilled (the names in the tagging tool look different)

File: src/download_model.py (hunk 1 above)
Comment (markus-hinsche) [reply src/download_model.py:74]: Global variables are often defined in config files or on the top of the module (also cross-check with our other code)

File: src/result_generation/height/height_ensemble.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import utils.inference as inference  # noqa: E402
+import utils.preprocessing as preprocessing  # noqa: E402
+
+
+class HeightFlowDeepEnsemble(HeightFlow):
+    def run_height_flow_deepensemble(self):
+        depthmaps = self.process_depthmaps()
+        prediction_list_one = []
+        model_paths = glob.glob(
+            '/app/models/deepensemble/*')
```
Comment (markus-hinsche) [inline src/result_generation/height/height_ensemble.py:20]:
```suggestion
        model_paths = glob.glob('/app/models/deepensemble/*')
```

**Review by markus-hinsche — APPROVED** [review]
LGTM

Few suggestions for improvements. Names visible in tagging tool should be updated according to acceptance criteria

File: src/download_model.py [hunk 3]
```diff
@@ -7,6 +7,15 @@
 
 REPO_DIR = Path(__file__).parents[1].absolute()
 
+RUN_IDS = ['q1-ensemble-warmup_1610544610_eb44bfe2', 'q1-ensemble-warmup_1610547587_7ca932c3',
```
Comment (gangooteli) [inline src/download_model.py:10]: Pls rename to ensemble run ids instead of just run ids

File: src/result_generation/height/height_ensemble.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+sys.path.append(str(Path(__file__).parents[1]))
+import utils.inference as inference  # noqa: E402
+import utils.preprocessing as preprocessing  # noqa: E402
+
+
+class HeightFlowDeepEnsemble(HeightFlow):
+    def run_height_flow_deepensemble(self):
+        depthmaps = self.process_depthmaps()
+        prediction_list_one = []
```
Comment (gangooteli) [inline src/result_generation/height/height_ensemble.py:18]: can change prediction_list_one to prediction_list

File: src/utils/inference.py [hunk 5]
```diff
@@ -73,3 +73,9 @@ def get_standing_laying_prediction_local(numpy_array):
 
 def get_depthmapmultiartifactlatefusion_height_predictions_local(numpy_array):
     return depthmapmultiartifactlatefusion_height_model.predict(numpy_array)
+
+
+def get_ensemble_height_predictions_local(model_path, numpy_array):
+    model_path += '/outputs/best_model.ckpt/'
```
Comment (gangooteli) [inline src/utils/inference.py:79]: I don't think code is correct.
If _model_path_ is args of _get_ensemble_height_predictions_local()_, 
why you are harcoding value in line 79 
_model_path += '/outputs/best_model.ckpt/'_

**Review by gangooteli — CHANGES_REQUESTED** [review]
Pls make suggested changes.

File: src/utils/inference.py (hunk 5 above)
Comment (gangooteli) [reply src/utils/inference.py:79]: Missed the + sign.
Resolved !!!

**Pull request merged**